sidechain/storage/src/storage.rs

```diff
@@ -613,83 +639,70 @@ mod test {
 		{
-			let mut sidechain_db = SidechainStorage::<SignedBlock>::new(path.clone()).unwrap();
+			let mut sidechain_db = get_storage(temp_dir.path().to_path_buf());
 			sidechain_db.shards.push(shard);
 			sidechain_db.last_blocks.insert(shard, last_block);
+
 			// when
 			let result = sidechain_db.verify_block_ancestry(&signed_block_two.block());
 
 			// then
 			assert!(!result);
 		}
-		// clean up
-		let _ = DB::destroy(&Options::default(), path).unwrap();
 	}
 
 	#[test]
 	fn verify_block_ancestry_returns_false_no_last_block_registered() {
-		// given
-		let path = PathBuf::from("verify_block_ancestry_returns_false_no_last_block_registered");
+		let temp_dir = create_temp_dir();
 		let shard = H256::from_low_u64_be(1);
 		let signed_block = create_signed_block(8, shard);
 		{
-			let mut sidechain_db = SidechainStorage::<SignedBlock>::new(path.clone()).unwrap();
+			let mut sidechain_db = get_storage(temp_dir.path().to_path_buf());
 			sidechain_db.shards.push(shard);
 			// when
 			let result = sidechain_db.verify_block_ancestry(&signed_block.block());
 
 			// then
 			assert!(!result);
 		}
-		// clean up
-		let _ = DB::destroy(&Options::default(), path).unwrap();
 	}
 
 	#[test]
 	fn verify_block_ancestry_returns_false_if_no_shard() {
-		// given
-		let path = PathBuf::from("verify_block_ancestry_returns_false_if_no_shard");
+		let temp_dir = create_temp_dir();
 		let shard = H256::from_low_u64_be(1);
 		let signed_block = create_signed_block(8, shard);
 		{
-			let sidechain_db = SidechainStorage::<SignedBlock>::new(path.clone()).unwrap();
-			// when
+			let sidechain_db = get_storage(temp_dir.path().to_path_buf());
 			let result = sidechain_db.verify_block_ancestry(&signed_block.block());
-
-			// then
 			assert!(!result);
 		}
-		// clean up
-		let _ = DB::destroy(&Options::default(), path).unwrap();
 	}
 
 	#[test]
 	fn add_block_to_batch_works_with_new_shard() {
 		// given
-		let path = PathBuf::from("add_block_to_batch_works_with_new_shard");
+		let temp_dir = create_temp_dir();
 		let shard = H256::from_low_u64_be(1);
 		let signed_block = create_signed_block(8, shard);
 		let mut new_shard = false;
 		{
-			let mut sidechain_db = SidechainStorage::<SignedBlock>::new(path.clone()).unwrap();
+			let mut sidechain_db = get_storage(temp_dir.path().to_path_buf());
 			let mut batch = WriteBatch::default();
 			assert!(batch.is_empty());
-			// when
-			sidechain_db.add_block_to_batch(&signed_block, &mut new_shard, &mut batch);
 
-			// then
+			sidechain_db
+				.add_block_to_batch(&signed_block, &mut new_shard, &mut batch)
+				.unwrap();
+
 			assert!(new_shard);
-			// ensure Writebatch is not empty anymore:
 			assert!(!batch.is_empty());
 		}
-		// clean up
-		let _ = DB::destroy(&Options::default(), path).unwrap();
 	}
 
 	#[test]
 	fn add_block_to_batch_does_not_add_shard_if_existent() {
-		// given
-		let path = PathBuf::from("add_block_to_batch_does_not_add_shard_if_existent");
+		let temp_dir = create_temp_dir();
```

Comment:
So the temporary directory cleans itself up when it goes out of scope?

Reply:
Yes, it does. 👍